compose.js

```diff
@@ -69,40 +121,50 @@ function createStamp(descriptor, composeFunction) {
  * @returns {object} Returns the dstDescriptor argument.
  */
 function mergeComposable(dstDescriptor, srcComposable) {
-  const srcDescriptor = (srcComposable && srcComposable.compose) || srcComposable;
+  // Check if it's a stamp or something else.
+  const srcDescriptor = isStamp(srcComposable) ? srcComposable.compose : srcComposable;
+  // Ignore everything but things we can merge.
   if (!isDescriptor(srcDescriptor)) return dstDescriptor;
 
   const combineProperty = (propName, action) => {
-    if (!isObject(srcDescriptor[propName])) return;
-    if (!isObject(dstDescriptor[propName])) dstDescriptor[propName] = {};
+    // Do not create destination properties if there is no need.
+    if (!isDescriptor(srcDescriptor[propName])) return;
+    // Check if the destination is malformed, fix the problem if any.
+    if (!isDescriptor(dstDescriptor[propName])) dstDescriptor[propName] = {};
+    // Deep merge or shallow assign objects.
     action(dstDescriptor[propName], srcDescriptor[propName]);
   };
 
   combineProperty('methods', assign);
   combineProperty('properties', assign);
-  combineProperty('deepProperties', merge);
+  combineProperty('deepProperties', mergeDescriptor);
   combineProperty('propertyDescriptors', assign);
   combineProperty('staticProperties', assign);
-  combineProperty('staticDeepProperties', merge);
+  combineProperty('staticDeepProperties', mergeDescriptor);
   combineProperty('staticPropertyDescriptors', assign);
   combineProperty('configuration', assign);
-  combineProperty('deepConfiguration', merge);
-  if (Array.isArray(srcDescriptor.initializers)) {
-    const initializers = (dstDescriptor.initializers || [])
-      .concat(srcDescriptor.initializers)
-      .filter(isFunction);
-    dstDescriptor.initializers = uniq(initializers);
+  combineProperty('deepConfiguration', mergeDescriptor);
+
+  if (isArray(srcDescriptor.initializers)) {
+    // Initializers must be concatenated. '.concat' will also create a new array instance.
+    const dstInitializers = (dstDescriptor.initializers || []).concat(srcDescriptor.initializers);
+    // The resulting initializers array must contain function only, and
+    // must not have duplicate initializers - the first occurrence wins.
+    dstDescriptor.initializers = uniq(dstInitializers.filter(isFunction));
   }
 
   return dstDescriptor;
 }
 
 /**
- * Given the list of composables (stamp descriptors and stamps) returns a new stamp (composable factory function).
+ * Given the list of composables (stamp descriptors and stamps)
+ * returns a new stamp (composable factory function).
```

Comment:
@boneskull
```js
/**
 * @typedef {Function} Stamp
 * @property {Function} compose
 */
```
So, where to get my beer? 😆 

But on the more serious note, give me a minute.